File: Search Matrix Problem/search_matrix.c
#include <stdio.h> 
int search(int mat[4][4], int n, int x) 
{ 
   int i=0,j=n-1;
   while(i < n && j >= 0)
   {
       if(mat[i][j] == x)
       {
           printf("\n Element Found at :- %d , %d",i,j);
           return 1;
       }
       if(mat[i][j] > x)
       {
           j--;
       }
       else
       {
           i++;
       }
   }
   printf("Element Not Found");
   return 0;
} 
int main() 
{ 
    int mat[4][4] = { 
        { 10, 20, 30, 40 }, 
        { 15, 25, 35, 45 }, 
        { 27, 29, 37, 48 }, 
        { 32, 33, 39, 50 }, 
    }; 
    search(mat, 4, 25); 
    return 0; 
}

File: Search Matrix Problem/search_matrix.php
<?php
function search(&$mat , $n ,$x)
{
    $i = 0;
    $j = $n-1;
    while($i < $n && $j >= 0)
    {
        if($mat[$i][$j] == $x)
        {
            echo "Element Found at :- ".$i.",".$j;
            return 1;
        }
        if($mat[$i][$j] > $x)
        {
            $j--;
        }
        else 
        {
            $i++;
        }
    }
    echo "\n Element Not Found";
    return 0;
}
$mat = array(array(10,20,30,40),
             array(15,25,35,45),
             array(27,29,37,48),
             array(31,33,39,50));
search($mat,4,37);
?>